CHANGELOG.md

```diff
@@ -3,6 +3,13 @@
 ## Unreleased
 
 ### Added
+
+- Add `EndLedger` in `GetEventsRequest`. This provides finer control and clarity on the range of ledgers being queried.
+- Disk-Based Event Storage: Events are now stored on disk instead of in memory. For context, storing approximately 3 million events will require around 1.5 GB of disk space.
+This change enhances the scalability and can now support a larger retention window (~7 days) for events.
+- Ledger Scanning Limitation: The getEvents RPC will now scan a maximum of `10,000` ledgers per request. This limits the resource usage and ensures more predictable performance, especially for queries spanning large ledger ranges.
+- A migration process has been introduced to transition event storage from in-memory to disk-based storage.
```

Comment:
We should boldly notify the user that upgrading to this version will take time to do the migration! At the top

Reply:
let me bold it but I dont think it will take that long honestly bc remember retention window is still at 1 day and not 7 days. We need to upgrade it all at once i.e. for Txns too in follow up PR. 

I have performed some sanity checks by running RPC locally, migration and ingestion works fine. Checked data in sqlite as well for events table

Waiting for @aditya1702 final confirmation as he was getting some weird error  while running RPC locally which I think was transient but all in all I am planning to merge this PR EOD